ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/InteractiveStatementExecutor.java

```diff
@@ -241,6 +254,13 @@ private void executePlan(
         new CommandStatus(CommandStatus.Status.EXECUTING, "Executing statement")
     );
     final ExecuteResult result = ksqlEngine.execute(serviceContext, configured);
+    long queryID = Long.MIN_VALUE;
+    if (greatestQueryId.isPresent() && mode == Mode.RESTORE) {
+      final String ltq = greatestQueryId.get().toString();
+      final int lastIndex = ltq.lastIndexOf("_");
+      queryID = Long.parseLong(ltq.substring(lastIndex + 1));
```

Comment:
this is breaking an abstraction barrier - we're relying on internal implementations of the string representation of the `QueryId` which doesn't guarantee that it will match the format that you assume here